project2/py/knowledge_base.py

```diff
@@ -22,3 +22,21 @@ def get_meat_substitute(self):
         index = random.randint(0, upper_bound)
         chosen_sub = self.meat_substitutes[index]
         return chosen_sub
+
+    def extract_methods(self, direction):
```

Comment:
I am not sure if this belongs in `knowledge_base.py`. I think we should only use the knowledge base to ask questions such as 'is this a meat?' or 'what are all the methods?' and leave the parsing (which is what it seems like this function is doing) elsewhere